ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -275,16 +285,29 @@ private struct BufferState<Value, Error: ErrorType> {
 	// producer was terminated.
 	var observers: Bag<Signal<Value, Error>.Observer>? = Bag()
 
-	// Appends a new value to the buffer, trimming it down to the given capacity
-	// if necessary.
-	mutating func addValue(value: Value, upToCapacity capacity: Int) {
-		values.append(value)
-
-		let overflow = values.count - capacity
-		if overflow > 0 {
-			values.removeRange(0..<overflow)
-		}
-	}
+    mutating func addValue(value: Value) {
+        if self.capacity == 1 {
+            self.values = [value]
+            return
+        }
+        if values.count == 0 {
+            values = Array<Value>(count: self.capacity * 2, repeatedValue: value)
+        }
+        values[index] = value
```

Comment:
The test suite crashes here (on my local environment with the latest `master`) in the test case: https://github.com/ReactiveCocoa/ReactiveCocoa/blob/17bd9528eb436cab5d2f388edd17b0e6982f46a8/ReactiveCocoaTests/Swift/SignalProducerLiftingSpec.swift#L1005-L1017